ogcapi-types/src/movingfeatures/temporal_primitive_geometry.rs

```diff
@@ -0,0 +1,493 @@
+use chrono::{DateTime, Utc};
+use geojson::{JsonObject, JsonValue, LineStringType, PointType, PolygonType};
+use serde::{
+    Deserialize, Deserializer, Serialize, Serializer,
+    de::{self, Error},
+};
+
+use crate::common::Crs;
+
+use super::{json_utils, trs::Trs};
+
+/// TemporalPrimitiveGeometry Object
+///
+/// A [TemporalPrimitiveGeometry](https://docs.ogc.org/is/19-045r3/19-045r3.html#tprimitive) object describes the
+/// movement of a geographic feature whose leaf geometry at a time instant is drawn by a primitive geometry such as a
+/// point, linestring, and polygon in the two- or three-dimensional spatial coordinate system, or a point cloud in the
+/// three-dimensional spatial coordinate system.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct TemporalPrimitiveGeometry {
+    #[serde(flatten)]
+    pub value: Value,
+    #[serde(default)]
+    pub interpolation: Interpolation,
+    // FIXME apparently specification of moving features CRS and TRS is different to common::CRS ?!
+    pub crs: Option<Crs>,
+    pub trs: Option<Trs>,
+    pub foreign_members: Option<JsonObject>,
+}
+
+impl TemporalPrimitiveGeometry {
+    pub fn new(value: Value) -> Self {
+        Self {
+            value,
+            interpolation: Interpolation::default(),
+            crs: Default::default(),
+            trs: Default::default(),
+            foreign_members: Default::default(),
+        }
+    }
+}
+
+impl<V> From<V> for TemporalPrimitiveGeometry
+where
+    V: Into<Value>,
+{
+    fn from(v: V) -> TemporalPrimitiveGeometry {
+        TemporalPrimitiveGeometry::new(v.into())
+    }
+}
+
+///The value specifies the variants of a TemporalPrimitiveGeometry object with constraints on the interpretation of the
+///array value of the "coordinates" member, the same-length "datetimes" array member and the optional members "base" and
+///"orientations".
+#[derive(Clone, Debug, PartialEq)]
+pub enum Value {
```

Comment:
Could this be modeled as an internally tagged enum so that serde can do the work without the need for custom serialization?